plugins/modules/podman_prune.py

```diff
@@ -0,0 +1,247 @@
+        type: dict
+    network:
+        description:
+            - Whether to prune networks.
+        type: bool
+        default: false
+    network_filters:
+        description:
+            - A dictionary of filter values used for selecting networks to delete.
+            - See L(the podman documentation,
+              https://docs.podman.io/en/latest/markdown/podman-network-prune.1.html#filter)
+              for more information on possible filters.
+        type: dict
+    system:
+        description:
+            - Wheter to prune unused pods, containers, image, networks and volume data
+        type: bool
+        default: false
+    system_all:
+        description:
+            - Wheter to prune all unused images, not only dangling images.
+        type: bool
+        default: false
+    system_volumes:
+        description:
+            - Wheter to prune volumes currently unused by any container.
+        type: bool
+        default: false
+    volume:
+        description:
+            - Whether to prune volumes.
+        type: bool
+        default: false
+    volume_filters:
+        description:
+            - A dictionary of filter values used for selecting volumes to delete.
+            - See L(the podman documentation,
+              https://docs.podman.io/en/latest/markdown/podman-volume-prune.1.html#filter)
+              for more information on possible filters.
+        type: dict
+'''
+
+EXAMPLES = r'''
+- name: Prune containers older than 24h
+  containers.podman.podman_prune:
+      containers: true
+      containers_filters:
+          # only consider containers created more than 24 hours ago
+          until: 24h
+
+- name: Prune everything
+  containers.podman.podman_prune:
+      system: true
+
+- name: Prune everything (including non-dangling images)
+  containers.podman.podman_prune:
+      system: true
+      system_all: true
+      system_volumes: true
+'''
+
+RETURN = r'''
+# containers
+containers:
+    description:
+      - List of IDs of deleted containers.
+    returned: I(containers) is C(true)
+    type: list
+    elements: str
+    sample: []
+
+# images
+images:
+    description:
+      - List of IDs of deleted images.
+    returned: I(images) is C(true)
+    type: list
+    elements: str
+    sample: []
+
+# networks
+networks:
+    description:
+      - List of IDs of deleted networks.
+    returned: I(networks) is C(true)
+    type: list
+    elements: str
+    sample: []
+
+# volumes
+volumes:
+    description:
+      - List of IDs of deleted volumes.
+    returned: I(volumes) is C(true)
+    type: list
+    elements: str
+    sample: []
+
+# system
+system:
+  description:
+    - List of ID of deleted containers, volumes, images, network and total reclaimed space
+  returned: I(system) is C(true)
+  type: list
+  elements: str
+  sample: []
+'''
+
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+def podmanExec(module, target, filters, executable):
```

Comment:
I'd recommend to to filters preparing in a separate function, and to leave here only podman execution.
Then you can jump earlier from filters function and avoid deep nesting.
Also filters processing is completely independent, so it's right to do it separately.